pkg/scheduler/objects/application.go

```diff
@@ -1267,6 +1270,7 @@ func (sa *Application) RemoveAllAllocations() []*Allocation {
 	// cleanup allocated resource for app (placeholders and normal)
 	sa.allocatedResource = resources.NewResource()
 	sa.allocatedPlaceholder = resources.NewResource()
+	sa.clearPlaceholderTimer()
```

Comment:
Please clear the state timer as well. This will be probably OK, but just to make sure it is cleared I would call a clear on that as well.